packages/opentelemetry-resources/src/utils.ts

```diff
@@ -14,8 +14,10 @@
  * limitations under the License.
  */
 
-export const isPromiseLike = <R>(val: any): val is PromiseLike<R> => {
+export const isPromiseLike = <R>(val: unknown): val is PromiseLike<R> => {
   return (
-    val !== null && typeof val === 'object' && typeof val.then === 'function'
+    val !== null &&
+    typeof val === 'object' &&
+    typeof (val as Partial<PromiseLike<R>>).then === 'function'
   );
```

Comment:
Adding `'then' in val` to the existing chain would also work (instead of doing the cast), and sometimes I prefer that for readability. However that does impose a tiny bit of additional runtime cost so I avoided it here.

Reply:
@david-luna resolved

For context: #5350 landed a similar fix so this PR isn't strictly necessary anymore, if the only goal is to get rid of the `any` (which #5350 did). This is slightly more precise and matches the intention of the code better.